gobblin-runtime/src/main/java/gobblin/runtime/AbstractJobLauncher.java

```diff
@@ -204,10 +202,6 @@ public void apply(JobListener jobListener, JobContext jobContext) throws Excepti
 
   @Override
   public void launchJob(JobListener jobListener) throws JobException {
-    if (this.jobContext.getJobMetricsOptional().isPresent()) {
```

Comment:
Annoyingly, these changes are going to cause #764 to not merge correctly.  Would be really nice to get #764 in as it was fixing a couple of other bugs.